### ISSUE TITLE
Dynamical changing of root_dir, conf_stack

### ISSUE THREAD

**Issue by jkoal** [MasonMcGill/artisan#4]
**Motivation:** - Dynamically processing artifacts from different/ several directories.
**Problem:** - Artifact creation does not actually respond to attempted changes to the conf.

**Example 1: push_conf**
Expected behaviour:
``` python
In [1]: root_dir = Path('/path/to/dir')  # specify a directory
In [2]: artisan.push_conf(root_dir=root_dir)
In [3]: foo = artisan.Artifact("bar")
In [4]: print(foo.path)
Out [4]: Path('/path/to/dir/bar')  # same directory as specified
```

Actual behaviour:
``` python
In [1]: root_dir = Path('/path/to/dir') # specify a directory
In [2]: artisan.push_conf(root_dir=root_dir)
In [3]: foo = artisan.Artifact("bar")
In [4]: print(foo.path)
Out [4]: Path('/current/working/dir/bar')  # still in working directory
```

**Example 2: context manager**
Expected behaviour:
``` python
In [1]: dir1 = Path('/path/to/dir1')
In [2]: with artisan.using_conf(root_dir=dir1):  # process artifacts in dir1
              foo1 = artisan.Artifact("bar")
              print(foo1.path)
Out [2]: Path('/path/to/dir1/bar')
In [3]: dir2 = Path('/path/to/dir2')  
In [4]: with artisan.using_conf(root_dir=dir2): # process artifacts in dir2
              foo2 = artisan.Artifact("bar")
              print(foo2.path)
Out [4]: Path('/path/to/dir2/bar')
```

Actual behaviour:
``` python
In [1]: dir1 = Path('/path/to/dir1')
In [2]: with artisan.using_conf(root_dir=dir1):  # process artifacts in dir1
              foo1 = artisan.Artifact("bar")
              print(foo1.path)
Out [2]: Path('/current/working/dir/bar')
In [3]: dir2 = Path('/path/to/dir2')  
In [4]: with artisan.using_conf(root_dir=dir2): # process artifacts in dir2
              foo2 = artisan.Artifact("bar")
              print(foo2.path)
Out [4]: Path('/current/working/dir/bar')
```

**Comment by MasonMcGill** [comment]
Hi Janne,

This is the intended behavior. Setting `root_dir` changes the default directory for artifact creation (when no path is specified), not the meaning of valid relative paths. *i.e.* it does not change the current working directory—I think that would be unnecessarily obfuscatory, since you can already use `os.chdir`.

If you'd like to create an artifact in the `root_dir`, you can use a "@/path/like/this". It looks like this is documented in the comments, but not the compiled docs, which should be fixed.

**Comment by MasonMcGill** [comment]
Resolved by 9d81941a86bf7bbe2bd80678f55f2015ac56a213
